### PR TITLE
Clean up E0205 explanation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Dylan-DPC

### CODE REVIEW

File: src/librustc_error_codes/error_codes/E0206.md [hunk 1]
```diff
@@ -1,12 +1,17 @@
-You can only implement `Copy` for a struct or enum. Both of the following
-examples will fail, because neither `[u8; 256]` nor `&'static mut Bar`
-(mutable reference to `Bar`) is a struct or enum:
+The `Copy` trait was implemented on a type which isn't a struct or an enum.
```
Comment (Dylan-DPC) [inline src/librustc_error_codes/error_codes/E0206.md]:
```suggestion
The `Copy` trait was implemented on a type which is neither a struct nor an enum.
```

File: src/librustc_error_codes/error_codes/E0206.md [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
+impl Copy for Foo { } // error!
 
 #[derive(Copy, Clone)]
 struct Bar;
-impl Copy for &'static mut Bar { } // error
+
+impl Copy for &'static mut Bar { } // error!
 ```
+
+You can only implement `Copy` for a struct or enum. Both of the previous
```
Comment (Dylan-DPC) [inline src/librustc_error_codes/error_codes/E0206.md]:
```suggestion
You can only implement `Copy` for a struct or an enum. Both of the previous
```

**Review by Dylan-DPC-zz — CHANGES_REQUESTED** [review]

**Comment by GuillaumeGomez** [comment]
Updated!

**Pull request merged**